bundles/framework/announcements/view/AnnouncementsCollapse.jsx

```diff
@@ -0,0 +1,70 @@
+import React from "react";
+import PropTypes from 'prop-types';
+import { Controller, LocaleConsumer } from 'oskari-ui/util';
+import { Message, Collapse, CollapsePanel } from 'oskari-ui';
+import { AnnouncementsModal } from './AnnouncementsModal';
+import { Divider } from 'antd';
+
+//Collapse panel -> set title, content and date range according to announcement
+
+const { Panel } = Collapse;
+
+const AnnouncementsCollapse = ({controller, updated, panels, modals, checked }) => {
+
+      if (!updated) {
+        controller.getAnnouncements(function(ann){
+          console.log("get announcements collapse");
+          console.log(ann);
+
+        
+          var panels = [];
+          var modals = [];
+            ann.data.forEach((announcement) => {
+
+              if (announcement.active && controller.showModal(announcement.id)) {
+                //if announcement is active, then show pop-up of the content
+                modals.push(announcement);
+              }
+              panels.push(
+                      <CollapsePanel header={announcement.title} key={announcement.id}>
+                          <h3>{announcement.title}</h3>
+                          <p>{announcement.content}</p>
+                          <Divider />
+                          <b><Message messageKey={'valid'} /></b>
+                          <p>{announcement.begin_date} - {announcement.end_date}</p>
+                      </CollapsePanel>
+                );
+            });
+            controller.updatePanelsModals(panels,modals);
+          });
+      }
+      
+      
+  return (
+    <div>
+      {modals.map((modal, index) => {
+            return (
+              <AnnouncementsModal
+              id={modal.id}
+              controller={controller}
+              title={modal.title}
+              content={modal.content}
+              key={modal.id}
+              index={index}
+              checked={checked}
+              />
+            );
+          })}
+      <div>
+        <Collapse accordion>{panels}</Collapse>
```

Comment:
Loop the announcements here so you don't need to push the panels in the state and have all of the render inside here without checking what is {panels} from props:
```suggestion
        <Collapse accordion>
          { announcements.map((announcement) => {
              return {
              <CollapsePanel header={announcement.title} key={announcement.id}>
                  <h3>{announcement.title}</h3>
                  <p>{announcement.content}</p>
                  <Divider />
                  <b><Message messageKey={'valid'} /></b>
                  <p>{announcement.begin_date} - {announcement.end_date}</p>
              </CollapsePanel>
            })}
        </Collapse>
```

Reply:
Will push these on Friday!